src/sage/dynamics/arithmetic_dynamics/projective_ds.py

```diff
@@ -395,7 +395,7 @@ def __classcall_private__(cls, morphism_or_polys, domain=None, names=None):
             # homogenize!
             f = morphism_or_polys
             aff_CR = f.parent()
-            if (not isinstance(aff_CR, PolynomialRing_general) and not isinstance(aff_CR, FractionField_generic)
+            if (not isinstance(aff_CR, PolynomialRing_generic) and not isinstance(aff_CR, FractionField_generic)
```

Comment:
```suggestion
            if (not isinstance(aff_CR, (PolynomialRing_generic, FractionField_generic))
```
should be the same, but that's really not important.